File: guild/src/main/java/com/os/operando/guild/Ennead.java
package com.os.operando.guild;

/**
 * A tuple of nine elements.
 *
 * @param <F>  first element type
 * @param <S>  second element type
 * @param <T>  third element type
 * @param <FO> fourth element type
 * @param <FI> five element type
 * @param <SI> six element type
 * @param <SE> seven element type
 * @param <E>  eight element type
 * @param <N> nine element type
 */
public class Ennead<F, S, T, FO, FI, SI, SE, E, N> {

    private final F first;
    private final S second;
    private final T third;
    private final FO fourth;
    private final FI five;
    private final SI six;
    private final SE seven;
    private final E eight;
    private final N nine;

    private Ennead(final F first, final S second, final T third, final FO fourth, final FI five, final SI six, final SE seven, final E eight, final N nine) {
        this.first = first;
        this.second = second;
        this.third = third;
        this.fourth = fourth;
        this.five = five;
        this.six = six;
        this.seven = seven;
        this.eight = eight;
        this.nine = nine;
    }

    public static <F, S, T, FO, FI, SI, SE, E, N> Ennead<F, S, T, FO, FI, SI, SE, E, N> create(F first, S second, T third, FO fourth, FI five, SI six, SE seven, E eight, N nine) {
        return new Ennead<>(first, second, third, fourth, five, six, seven, eight, nine);
    }

    public F getFirst() {
        return first;
    }

    public S getSecond() {
        return second;
    }

    public T getThird() {
        return third;
    }

    public FO getFourth() {
        return fourth;
    }

    public FI getFive() {
        return five;
    }

    public SI getSix() {
        return six;
    }

    public SE getSeven() {
        return seven;
    }

    public E getEight() {
        return eight;
    }

    public N getNine() {
        return nine;
    }
}

File: guild-kotlin/src/main/kotlin/com/os/operando/guild/kt/Ennead.kt
package com.os.operando.guild.kt

/**
 * A tuple of eight elements.
 *
 * @param F  first element type
 * @param S  second element type
 * @param T  third element type
 * @param FO fourth element type
 * @param FI five element type
 * @param SI six element type
 * @param SE seven element type
 * @param E eight element type
 * @param N nine element type
 * @property first First value
 * @property second Second value
 * @property third Third value
 * @property fourth Fourth value
 * @property five Five value
 * @property six Six value
 * @property seven Seven value
 * @property eight Eight value
 * @property nine Nine value
 */
data class Ennead<out F, out S, out T, out FO, out FI, out SI, out SE, out E, out N>(
        val first: F,
        val second: S,
        val third: T,
        val fourth: FO,
        val five: FI,
        val six: SI,
        val seven: SE,
        val eight: E,
        val nine: N) {
}